server/src/dependency.rs

```diff
@@ -0,0 +1,169 @@
+use {
+    move_binary_format::errors::PartialVMError,
+    moved::{
+        block::{BlockHash, BlockQueries, BlockRepository, MovedBlockHash},
+        move_execution::{BaseTokenAccounts, MovedBaseTokenAccounts},
+        receipt::{ReceiptQueries, ReceiptRepository},
+        transaction::{TransactionQueries, TransactionRepository},
+    },
+    moved_genesis::config::GenesisConfig,
+    moved_state::State,
+};
+
+#[cfg(feature = "storage")]
+pub type SharedStorage = &'static moved_storage::RocksDb;
+#[cfg(not(feature = "storage"))]
+pub type SharedStorage = moved::in_memory::SharedMemory;
+#[cfg(feature = "storage")]
+pub type ReceiptStorage = &'static moved_storage::RocksDb;
+#[cfg(not(feature = "storage"))]
+pub type ReceiptStorage = moved::receipt::ReceiptMemory;
+#[cfg(feature = "storage")]
+pub type StateQueries = moved::state_actor::InMemoryStateQueries;
+#[cfg(not(feature = "storage"))]
+pub type StateQueries = moved::state_actor::InMemoryStateQueries;
+
+pub fn block_hash() -> impl BlockHash + Send + Sync + 'static {
+    MovedBlockHash
+}
+
+pub fn base_token(
+    genesis_config: &GenesisConfig,
+) -> impl BaseTokenAccounts + Send + Sync + 'static {
+    MovedBaseTokenAccounts::new(genesis_config.treasury)
+}
+
+pub fn memory() -> SharedStorage {
+    #[cfg(feature = "storage")]
+    {
+        db()
+    }
+    #[cfg(not(feature = "storage"))]
+    {
+        moved::in_memory::SharedMemory::new()
+    }
+}
+
+pub fn block_repository() -> impl BlockRepository<Storage = SharedStorage> + Send + Sync + 'static {
+    #[cfg(feature = "storage")]
+    {
+        moved_storage::block::RocksDbBlockRepository
+    }
+    #[cfg(not(feature = "storage"))]
+    {
+        moved::block::InMemoryBlockRepository::new()
+    }
+}
+
+pub fn state() -> impl State<Err = PartialVMError> + Send + Sync + 'static {
+    moved_state::InMemoryState::new()
+}
+
+pub fn state_query(genesis_config: &GenesisConfig) -> StateQueries {
+    #[cfg(feature = "storage")]
+    {
+        moved::state_actor::InMemoryStateQueries::from_genesis(genesis_config.initial_state_root)
```

Comment:
Is this supposed to be `moved_sorage::state::RocksDbState`?

Reply:
Currently there is no such implementation.

But it is not strictly required, it only stores a single 32-byte hash per block.

Reply:
Maybe we should make this change too while we're at it. I noticed the CPU performance issue lingering still. We just want to completely rule out lack of storage causing it.

Reply:
You're right the `moved_storage::state::RocksDbState` was not actually used as it should be.

I just added it.